Doc/library/multiprocessing.rst

```diff
@@ -2127,6 +2128,16 @@ with the :class:`Pool` class.
    Note that the methods of the pool object should only be called by
    the process which created the pool.
 
+   .. warning::
+      :class:`multiprocessing.pool` objects have internal resources that need to be
+      properly managed (like any other resource) by using the pool as a context manager
+      or by calling :meth:`close` and :meth:`terminate` manually. Failure to do this
+      can lead to the process hanging on finalization.
+
+      Note that is **not correct** to rely on the garbage colletor to destroy the pool
+      as CPython does not assure that the finalizer of the pool will be called
+      (see :meth:`object.__del__` for more information).
```

Comment:
My potential fix is to use `atexit.register` when we can't use the context manager. Worth suggesting here? (Also let me know if it's clearly a bad idea for some reason!)

Reply:
> My potential fix is to use `atexit.register` when we can't use the context manager. Worth suggesting here? (Also let me know if it's clearly a bad idea for some reason!)

That works, but the problem is that is still a lossy method of resource managing (you would not use that method for all your file objects for instance). Technically, someone that is not "the interpreter" should own the pool and should be in charge of destroying it correctly.

I understand that some architecture and designs make this challenging so things like `atexit` do work for those cases, but I think is better not to explicitly mention that here so users don't start to use that as a silver bullet.

What do you think?

Reply:
Yeah that makes sense. Otherwise, text LGTM and reads nicely in the rendered artifact!